src/main/java/application/rest/v1/ResourcesEndpoint.java

```diff
@@ -0,0 +1,1798 @@
+/*
+ * Copyright 2019 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package application.rest.v1;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.validation.constraints.Pattern;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.eclipse.microprofile.openapi.annotations.Operation;
+import org.eclipse.microprofile.openapi.annotations.parameters.Parameter;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponses;
+import org.eclipse.microprofile.openapi.annotations.tags.Tag;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.ibm.kappnav.logging.Logger;
+
+import application.rest.v1.configmaps.ConfigMapProcessor;
+import application.rest.v1.configmaps.SectionConfigMapProcessor;
+import io.kubernetes.client.ApiClient;
+import io.kubernetes.client.ApiException;
+import io.kubernetes.client.apis.CustomObjectsApi;
+import io.kubernetes.client.apis.CoreV1Api;
+import io.kubernetes.client.models.V1Secret;
+
+@Path("/resources")
+@Tag(name = "resources", description="kAppNav Resources API")
+public class ResourcesEndpoint extends KAppNavEndpoint {
+    private static final String className = ResourcesEndpoint.class.getName();
+
+    //common constants
```

Comment:
The design of this API depends on a static list of known kinds.  This is a problem for these reasons:

1. It cannot list resource of custom resource definition it does not know about.  There are many that get installed optionally - e.g. knative.   This API must handle unknown custom resources.  

1. It is out of synch with ComponentInfoRegistry.  The reason that is a problem is because this API will be used to help users select components to add to an application.  If those components are unknown to ComponentInfoRegistry, they will not be returned by the Components API.  This API and ComponentInfoRegistry must support the same resource kinds.  It might be better to extend ComponentInfoRegistry and call it from this API.